File: app/src/main/java/io/jeffchang/buttondemo/ui/userlist/UserListFragmentProvider.java
package io.jeffchang.buttondemo.ui.userlist;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import io.jeffchang.buttondemo.ui.userlist.view.UserListFragment;

/**
 * Provides dependencies into the list fragment.
 */

@Module
public abstract class UserListFragmentProvider {

    @ContributesAndroidInjector(modules = {UserListFragmentModule.class})
    abstract UserListFragment provideListFragment();
}

File: app/src/main/java/io/jeffchang/buttondemo/ui/userlist/UserListFragmentProvider.kt
package io.jeffchang.buttondemo.ui.userlist

import dagger.Module
import dagger.android.ContributesAndroidInjector
import io.jeffchang.buttondemo.ui.userlist.view.UserListFragment

/**
 * Provides dependencies into the list fragment.
 */

@Module
abstract class UserListFragmentProvider {

    @ContributesAndroidInjector(modules = [UserListFragmentModule::class])
    abstract fun provideListFragment(): UserListFragment
}